### ISSUE TITLE
google_dns_record_set NS create fails on apex

### ISSUE THREAD

**Issue by matti** [terraform-providers/terraform-provider-google#836]
with

```
resource "google_dns_managed_zone" "apex" {
  name     = "mydomain-com"
  dns_name = "mydomain.com."
}

resource "google_dns_managed_zone" "sub" {
  name     = "sub-mydomain-com"
  dns_name = "sub.mydomain.com."
}

resource "google_dns_record_set" "apex_sub_ns" {
  name = "sub.mydomain.com."
  type = "NS"
  ttl  = 300

  managed_zone = "${google_dns_managed_zone.apex.name}"

  rrdatas = ["${google_dns_managed_zone.sub.name_servers}"]
}
```

I get

```
* google_dns_record_set.apex_sub_ns: Error creating DNS RecordSet: googleapi: Error 400: The resource record set 'entity.change.deletions[0]' is invalid because a zone must contain exactly one resource record set of type 'NS' at the apex., invalidZoneApex
```

but if I change the "NS" type to "TXT", apply and then change it back to "NS" it works.

**Comment by rosbo** [comment]
Hi Matti,

@paddycarver fixed this bug in #807. 

The next release will include a fix for this bug.

Thanks.

**Comment by paddycarver** [comment]
That release just went out! `terraform init -upgrade` should fix this. If it doesn't, feel free to reply to this issue and we can take a look.
